unified-scheduler-pool/src/lib.rs

```diff
@@ -194,34 +199,422 @@ impl TaskHandler for DefaultTaskHandler {
+            handler_threads: Vec::with_capacity(handler_count),
+            accumulator_thread: None,
+        }
+    }
+
+    fn execute_task_with_handler(
+        bank: &Arc<Bank>,
+        executed_task: &mut Box<ExecutedTask>,
+        handler_context: &HandlerContext,
+    ) {
+        debug!("handling task at {:?}", thread::current());
+        TH::handle(
+            &mut executed_task.result_with_timings.0,
+            &mut executed_task.result_with_timings.1,
+            bank,
+            executed_task.task.transaction(),
+            executed_task.task.task_index(),
+            handler_context,
+        );
+    }
+
+    fn propagate_context_to_handler_threads(
+        runnable_task_sender: &mut Sender<ChainedChannel<Task, SchedulingContext>>,
+        context: SchedulingContext,
+        handler_count: usize,
+    ) {
+        let (next_sessioned_task_sender, runnable_task_receiver) = unbounded();
+        for _ in 0..handler_count {
+            runnable_task_sender
+                .send(ChainedChannel::chain_to_new_channel(
+                    context.clone(),
+                    runnable_task_receiver.clone(),
+                ))
+                .unwrap();
+        }
+        *runnable_task_sender = next_sessioned_task_sender;
+    }
+
+    fn take_session_result_with_timings(&mut self) -> ResultWithTimings {
+        self.session_result_with_timings.take().unwrap()
+    }
+
+    fn put_session_result_with_timings(&mut self, result_with_timings: ResultWithTimings) {
+        assert_matches!(
+            self.session_result_with_timings
+                .replace(result_with_timings),
+            None
+        );
+    }
+
+    fn start_threads(&mut self, context: &SchedulingContext) {
+        let (runnable_task_sender, runnable_task_receiver) =
+            unbounded::<ChainedChannel<Task, SchedulingContext>>();
+        let (executed_task_sender, executed_task_receiver) = unbounded::<ExecutedTaskPayload>();
+        let (finished_task_sender, finished_task_receiver) = unbounded::<Box<ExecutedTask>>();
+        let (accumulated_result_sender, accumulated_result_receiver) =
+            unbounded::<Option<ResultWithTimings>>();
+
+        let mut result_with_timings = self.session_result_with_timings.take();
+
+        // High-level flow of new tasks:
+        // 1. the replay stage thread send a new task.
+        // 2. the scheduler thread accepts the task.
+        // 3. the scheduler thread dispatches the task after proper locking.
+        // 4. the handler thread processes the dispatched task.
+        // 5. the handler thread reply back to the scheduler thread as an executed task.
+        // 6. the scheduler thread post-processes the executed task.
+        // 7. the scheduler thread send the executed task to the accumulator thread.
+        // 8. the accumulator thread examines the executed task's result and accumulate its timing,
+        //    finally dropping the transaction inside the executed task.
```

Comment:
across the new code, i used the term task instead of transaction for envisioning bundles as well along with transactions under the generic struct of Task.....